apps/src/templates/ImageWithStatus.jsx

```diff
@@ -0,0 +1,59 @@
+import React, {Component, PropTypes} from 'react';
+
+/**
+ * @fileoverview Wraps an <img> with a <div> that indicates the status of the
+ * image being loaded. The purpose of this is to be able to wait until images
+ * have finished loading within tests.
+ */
+
+/** @enum {string} */
+export const STATUS = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error',
+};
+
+export class ImageWithStatus extends Component {
+  static propTypes = {
+    src: PropTypes.string.isRequired,
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      imageStatus: STATUS.LOADING,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.src !== nextProps.src) {
+      this.setState({imageStatus: STATUS.LOADING});
+    }
+  }
+
+  handleImageLoad() {
+    this.setState({imageStatus: STATUS.LOADED});
+  }
+
+  handleImageError() {
+    this.setState({imageStatus: STATUS.ERROR});
+  }
+
+  render() {
+    return (
+      <div>
+        <img
+          src={this.props.src}
+          width={this.props.width}
+          height={this.props.height}
+          onLoad={this.handleImageLoad.bind(this)}
+          onError={this.handleImageError.bind(this)}
```

Comment:
You're better off doing your binding in the constructor (so that it doesnt happen again on each render). i.e
```
this.handleImageLoad = this.handleImageLoad.bind(this);
```